pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -180,3 +184,90 @@ type HTTPSDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// ConsulSDConfig defines a Consul service discovery configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#consul_sd_config
+// +k8s:openapi-gen=true
+type ConsulSDConfig struct {
+	// A valid string consisting of a hostname or IP followed by an optional port number.
+	// +kubebuilder:validation:MinLength:=1
+	// +required
+	Server string `json:"server"`
+	// Consul ACL TokenRef, If not provided it will use the ACL from the local Consul Agent.
+	// +optional
+	TokenRef *corev1.SecretKeySelector `json:"tokenRef,omitempty"`
+	// Consul Datacenter name, If not provided it will use the local Consul Agent Datacenter.
+	// +optional
+	Datacenter *string `json:"datacenter,omitempty"`
+	// Namespaces are only supported in Consul Enterprise.
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+	// Admin Partitions are only supported in Consul Enterprise.
+	// +optional
+	Partition *string `json:"partition,omitempty"`
+	// HTTP Scheme default "http"
+	// +kubebuilder:validation:Enum=http;https
```

Comment:
Kube API conventions recommend upper-case for acronyms.
```suggestion
	// +kubebuilder:validation:Enum=HTTP;HTTPS
```